### PR TITLE
Fused elementwises kernels and ops

### PR DESCRIPTION
**Pull request by HulekJakub**
### PR types

Others
### PR changes

Others
### Description

- added fused elementwise kernel
- added fused elementwises ops (add, sub, mul, div)
- mkldnn -> onednn
- adjusted unit tests

### CODE REVIEW

File: paddle/fluid/framework/ir/CMakeLists.txt
Comment (Silv3S) [inline paddle/fluid/framework/ir/CMakeLists.txt]: As we change it already, please change `elt` to either `eltwise` or `elementwise` as it's more common among other passes.

File: paddle/fluid/operators/compat/fused_elementwise_add.pbtxt
Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_add.pbtxt]: `x_data_format` and `y_data_format` seem to be deprecated attributes. Please don't add them here. They should be also removed from base operators

File: paddle/fluid/operators/compat/fused_elementwise_add.pbtxt
Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_add.pbtxt]:
```suggestion
    name: "scale_x"
```
All attributes should be lowercase. We must unify it, as there are currently too many duplicates with quantization scales.

File: paddle/fluid/operators/compat/fused_elementwise_add.pbtxt
Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_add.pbtxt]:
```suggestion
  attrs {
    name: "fuse_activation"
    type: STRING
  }
  attrs {
    name: "fuse_alpha"
    type: FLOAT
  }
  attrs {
    name: "fuse_beta"
    type: FLOAT
  }
```

File: paddle/fluid/operators/compat/fused_elementwise_add.pbtxt
Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_add.pbtxt]:
```suggestion
```

File: paddle/fluid/operators/compat/fused_elementwise_add.pbtxt
Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_add.pbtxt:50]: There is also float `fused_output_scale` from _operator_scale_onednn_fuse_pass_

Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_add.pbtxt:50]: All comments to `fused_elementwise_add.pbtxt` are valid for other `fused_elementwise_.pbtxt`

File: paddle/fluid/operators/compat/fused_elementwise_mul.pbtxt
Comment (Silv3S) [inline paddle/fluid/operators/compat/fused_elementwise_mul.pbtxt]: `elementwise_mul` can have `fused_unsqueeze2_axes` attribute from`operator_unsqueeze2_onednn_fuse_pass`

**Review by Silv3S — DISMISSED** [review]
LGTM

**Comment by HulekJakub** [comment]
@Aurelius84 Could you approve operators not supporting double? OneDNN ops support only float, bf16, uint8 and int8

**Comment by HulekJakub** [comment]
@kolinwei Could you approve changes in test names? Coverage thinks they are deleted

**Comment by HulekJakub** [comment]
@raindrops2sea Could you approve PR size?

**Comment by HulekJakub** [comment]
@chenwhql Could you approve change to op_compat.yml? I deleted outdated data_format arguments

**Review by Silv3S — DISMISSED** [review]

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_placement_pass.cc
Comment (Silv3S) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_placement_pass.cc:37]: Please remove if-block cpu_quantize_placement_pass.cc:90
`// Remove this condition when all fused_elementwise ops are merged`
It was temporary solution, which is solved by your PR

File: paddle/fluid/operators/ops_extra_info.h
Comment (Silv3S) [inline paddle/fluid/operators/ops_extra_info.h]:
```suggestion
```

File: paddle/fluid/framework/ir/mkldnn/operator_unsqueeze2_onednn_fuse_pass.cc
Comment (Silv3S) [inline paddle/fluid/framework/ir/mkldnn/operator_unsqueeze2_onednn_fuse_pass.cc]:
```suggestion
```
Done in line 90 already

**Comment by HulekJakub** [comment]
@YuanRisheng Conflicts were resolved and last improvements were made. Could you take care of approving this?

**Review by YuanRisheng — DISMISSED** [review]

File: paddle/fluid/framework/ir/graph_pattern_detector.cc
Comment (zyfncg) [inline paddle/fluid/framework/ir/graph_pattern_detector.cc:2263]: Can `Scale_out` be deleted?

Comment (HulekJakub) [reply paddle/fluid/framework/ir/graph_pattern_detector.cc:2263]: There isn't a consistent naming scheme and many operators have "Scale" and many others have "scale", so it can be deleted once all operators have the same capitalization scheme, but it is outside of this PR scope

**Review by zyfncg — DISMISSED** [review]

**Review by Aurelius84 — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]
LGTM

File: python/paddle/incubate/autograd/primrules.py
Comment (XiaoguangHu01) [inline python/paddle/incubate/autograd/primrules.py:134]: 这里为什么删除这段逻辑呢？

**Review by JiabinYang — DISMISSED** [review]
LGTM

File: python/paddle/incubate/autograd/primrules.py
Comment (HulekJakub) [reply python/paddle/incubate/autograd/primrules.py:134]: This is basic elementwise test, which doesn't have scale arguments anymore, so that if will never happen

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Comment by YuanRisheng** [comment]
This PR is already out of data, you need merge the latest code into this PR for CI-Coverage @HulekJakub

**Review by Silv3S — DISMISSED** [review]

**Comment by HulekJakub** [comment]
@YuanRisheng PR was finally fixed. Could you approve again?

**Review by Silv3S — DISMISSED** [review]

**Review by YuanRisheng — DISMISSED** [review]

**Review by YuanRisheng — DISMISSED** [review]

**Review by zyfncg — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]
LGTM

**Review by Aurelius84 — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Comment by HulekJakub** [comment]
I had to modify code once again since tests were timing out sometimes on CI. I didn't happen on my desktop. Sorry for inconvenience, but could I ask you (@XiaoguangHu01, @Aurelius84, @XieYunshen, @zyfncg, @YuanRisheng) for approval once again?

**Review by YuanRisheng — DISMISSED** [review]

**Review by zyfncg — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]

**Review by Aurelius84 — DISMISSED** [review]

**Review by Silv3S — DISMISSED** [review]

**Comment by YuanRisheng** [comment]
I am sorry to inform you that you need update code again. I have analyzed the CI-Coverage and found the problem. The test `test_elementwise_add_mkldnn_op` has some issues and it is forbidden before. In this PR,   `test_elementwise_add_mkldnn_op ` is renamed to `test_elementwise_add_onednn_op `, it will be no longer be banned. So , you could change name to `test_elementwise_add_mkldnn_op` for passing it. Good Luck! @HulekJakub

**Comment by HulekJakub** [comment]
> 

@YuanRisheng Done. I've seen that PR-CI-Coverage first checks approves and fails if there aren't required ones, and only once the approves are there it runs tests, which really slows development (tests work on local, have to wait for approves, test doesn't work on CI, approves get dismissed which is just a waste of time)

**Comment by YuanRisheng** [comment]
> @YuanRisheng Done. I've seen that PR-CI-Coverage first checks approves and fails if there aren't required ones, and only once the approves are there it runs tests, which really slows development (tests work on local, have to wait for approves, test doesn't work on CI, approves get dismissed which is just a waste of time)

@HulekJakub Thank you for your feedback. I will report this issue to my team. And CI-Windows fails because of `set_tests_properties(test_elementwise_add_onednn_op PROPERTIES TIMEOUT 60)` in CMakeLists.txt

**Review by XieYunshen — APPROVED** [review]

**Review by YuanRisheng — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

**Review by Aurelius84 — APPROVED** [review]

**Review by xinyu-intel — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
@HulekJakub 

### You're Invited: Claim Your Custom PaddlePaddle Contributor Badge!
Baidu Open Source Office is looking for those of you who have quietly contributed to PaddlePaddle. We’ve prepared a special token of appreciation just for you—a custom commemorative badge engraved with your name.

image


Over the years, every step forward PaddlePaddle has taken has been made possible by your code, your passion, and your youth. Maybe you once submitted a line of code late at night, maybe you kindly answered a question in the community, or maybe it’s been a while since you last contributed…
But we’ve never forgotten you.

This time, we’re not just looking to reconnect with your code—we want to relive those years we burned brightly together.
Come join us—let’s once again—“reclaim our youth”!

If you’d like to take part, please send your **name, phone number, and a brief self-recommendation** to:
📧 [ext_paddle_oss@baidu.com](https://mailto:ext_paddle_oss@baidu.com/)
⏰ Deadline: **September 10th, 10:00 AM**
📌 Feel free to write your self-recommendation in your own style～
💡 No committee referral or repository write access is required—every past contributor is warmly welcome to join!

**Pull request merged**
